core/App/screens/CredentialDetails.tsx

```diff
@@ -21,30 +20,84 @@ import { useTheme } from '../contexts/theme'
 import { BifoldError } from '../types/error'
 import { CredentialMetadata } from '../types/metadata'
 import { CredentialStackParams, Screens } from '../types/navigators'
+import { CardLayoutOverlay, CardOverlayType, MetaOverlay, OCACredentialBundle } from '../types/oca'
 import { Field } from '../types/record'
 import { RemoveType } from '../types/remove'
+import { credentialTextColor, isValidIndyCredential, toImageSource } from '../utils/credential'
 import { getCredentialConnectionLabel } from '../utils/helpers'
+import { testIdWithKey } from '../utils/testable'
 
 type CredentialDetailsProps = StackScreenProps<CredentialStackParams, Screens.CredentialDetails>
 
+const paddingHorizontal = 24
+const paddingVertical = 16
+const logoHeight = 80
+
 const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route }) => {
   if (!route?.params) {
     throw new Error('CredentialDetails route prams were not set properly')
   }
 
   const { credentialId } = route?.params
+
   const { agent } = useAgent()
   const { t, i18n } = useTranslation()
+  const { TextTheme, ColorPallet } = useTheme()
+  const { OCABundle } = useConfiguration()
+
   const [, dispatch] = useStore()
   const [isRevoked, setIsRevoked] = useState<boolean>(false)
   const [revocationDate, setRevocationDate] = useState<string>('')
-  const [fields, setFields] = useState<Field[]>([])
-  const [isRevokedMessageHidden] = useState<boolean>(false)
   const [isRemoveModalDisplayed, setIsRemoveModalDisplayed] = useState<boolean>(false)
+  const [isRevokedMessageHidden, setIsRevokedMessageHidden] = useState<boolean>(false)
+
+  const [overlay, setOverlay] = useState<{
```

Comment:
Is this a common type or just a one off? Asking in case it should be created as a type.

Reply:
Hard to say yet if it should be a common type. Thinking maybe it should.